### PR TITLE
Adds ZeroMQ and the C++ bindings for ZeroMQ

### PR DESCRIPTION
**Pull request by tjflexmaster**

### CODE REVIEW

File: projects/zeromq.cmake [hunk 1]
```diff
@@ -0,0 +1,37 @@
+########################################
+# novas 
+########################################
+xpProOption(zeromq)
+set(ZEROMQ_VERSION 4.2.2)
+set(ZEROMQ_SRC_PATH ${CMAKE_BINARY_DIR}/xpbase/Source/zeromq)
+set(ZEROMQ_DLURL https://github.com/zeromq/libzmq/archive/v${ZEROMQ_VERSION}.tar.gz)
+# 4.2.1 Hash = 820cec2860a72c3257881a394d83bfc0
```
Comment (bnhhughes) [inline projects/zeromq.cmake]: This is probably ok to take out now if 4.2.2 is the version that is going to be used.

File: projects/README.md [hunk 2]
```diff
@@ -12,4 +12,6 @@
...
 |[Qt](http://qt.io/ 'Qt - Home')|[LGPL](http://www.qt.io/qt-licensing-terms/ 'LGPL')|One Qt Code: Create Powerful Applications & Devices|5.8.0|[repo](http://code.qt.io/cgit/qt/qt5.git 'Qt5 main repo')|[diff](https://github.com/distributePro/ovsrpro/blob/master/patches/qt5.patch 'patch/diff')|
 |[Qwt](http://http://qwt.sourceforge.net/ 'Qwt - Qt Widgets for Technical Applications')|[LGPL](http://qwt.sourceforge.net/qwtlicense.html 'LGPL with exceptions')|The Qwt library contains GUI Components and utility classes which are primarily useful for programs with a technical background.|6.1.2|none|[diff](https://github.com/distributePro/ovsrpro/blob/master/patches/qwtconfig.pri.patch 'patch/diff')|
+|[ZeroMQ](https://zeromq.org/ 'ZeroMQ - Home")|[LGPL](https://github.com/zeromq/libzmq/blob/master/README.md/ 'LGPL v3')|ZeroMQ is a lightweight messaging kernel that extends standard socket interfaces with more powerful features.|4.2.1|[repo](https://github.com/zeromq/libzmq.git)|none|
```
Comment (bnhhughes) [inline projects/README.md]: Below it looks like version 4.2.2 is the one that is actually being built - this version should be updated to match

File: projects/zeromq.cmake [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+set(ZEROMQ_DLMD5 18599afc000dacd4d5715abbb18bfe6b)
+set(PRO_ZEROMQ
+  NAME zeromq
+  WEB "ZEROMQ" https://zeromq.org "ZEROMQ"
+  LICENSE "open" https://github.com/zeromq/libzmq/blob/master/README.md "(See License Section. GNU Lesser General Public LIcense.)"
+  DESC "ZEROMQ is a lightweight messaging kernel is a library that extends standard socket interfaces."
+  VER ${ZEROMQ_VERSION}
+  DLURL ${ZEROMQ_DLURL}
+  DLMD5 ${ZEROMQ_DLMD5}
+  #PATCH ${PATCH_DIR}/zeromq.patch
```
Comment (bnhhughes) [inline projects/zeromq.cmake]: No longer needed?  I'm guessing that was only if we stuck with version 4.2.1?

**Comment by bnhhughes** [comment]
Just a couple of basic changes that should be made first before merging

File: projects/cppzmq.cmake [hunk 4]
```diff
@@ -0,0 +1,43 @@
+########################################
+# novas
```
Comment (brobeson) [inline projects/cppzmq.cmake]: Copy & paste

File: projects/zeromq.cmake [hunk 5]
```diff
@@ -0,0 +1,35 @@
+########################################
+# novas
```
Comment (brobeson) [inline projects/zeromq.cmake]: Copy & paste

**Review by brobeson — APPROVED** [review]

**Pull request merged**